### ISSUE TITLE
journald: provide a way to rotate and vacuum *all* journal files, including active ones

### ISSUE THREAD

**Issue by kotnik** [systemd/systemd#1017]
In a nutshell, this is the issue:

```
[root@carpo ~]# journalctl --list-boots
-7 53098887dd3b434e84482c78fe368b74 Mon 2012-10-29 20:54:44 CET—Mon 2012-10-29 20:54:45 CET
-6 6b92a21d70e849bf810706e19a759317 Tue 2012-10-30 09:16:39 CET—Tue 2012-10-30 09:16:39 CET
-5 70ca24cb88d34ccd8cc2be2d8488c0da Wed 2012-10-31 09:20:55 CET—Wed 2012-10-31 09:25:49 CET
-4 79bc28a557ff44c48867e230563200cd Tue 2012-12-25 22:48:08 CET—Tue 2012-12-25 22:48:08 CET
-3 d598ae81f61e4461830999733a5521c0 Sat 2015-08-22 01:38:06 CEST—Sat 2015-08-22 01:38:06 CEST
-2 22bbfc5414ee4526b876fe9aee54a365 Sat 2015-08-22 12:35:07 CEST—Sat 2015-08-22 23:24:23 CEST
-1 abde94a5c6444944afcb78ae2304dfd1 Sun 2015-08-23 17:21:46 CEST—Mon 2015-08-24 09:26:20 CEST
 0 eb8e1df90d8843fd88403635c9ad69f3 Mon 2015-08-24 09:42:41 CEST—Mon 2015-08-24 15:10:27 CEST
[root@carpo ~]# journalctl --vacuum-time=1months
Vacuuming done, freed 0B of archived journals on disk.
[root@carpo ~]# journalctl --list-boots
-7 53098887dd3b434e84482c78fe368b74 Mon 2012-10-29 20:54:44 CET—Mon 2012-10-29 20:54:45 CET
-6 6b92a21d70e849bf810706e19a759317 Tue 2012-10-30 09:16:39 CET—Tue 2012-10-30 09:16:39 CET
-5 70ca24cb88d34ccd8cc2be2d8488c0da Wed 2012-10-31 09:20:55 CET—Wed 2012-10-31 09:25:49 CET
-4 79bc28a557ff44c48867e230563200cd Tue 2012-12-25 22:48:08 CET—Tue 2012-12-25 22:48:08 CET
-3 d598ae81f61e4461830999733a5521c0 Sat 2015-08-22 01:38:06 CEST—Sat 2015-08-22 01:38:06 CEST
-2 22bbfc5414ee4526b876fe9aee54a365 Sat 2015-08-22 12:35:07 CEST—Sat 2015-08-22 23:24:23 CEST
-1 abde94a5c6444944afcb78ae2304dfd1 Sun 2015-08-23 17:21:46 CEST—Mon 2015-08-24 09:26:20 CEST
 0 eb8e1df90d8843fd88403635c9ad69f3 Mon 2015-08-24 09:42:41 CEST—Mon 2015-08-24 15:10:27 CEST
```

Please let me know if you need any additional information.

**Comment by poettering** [comment]
Hmm, most likely the journal files in question used an older naming scheme, and our clean-up logic doesn't recognize them as old. Could you run `ls -al /var/log/journal/*/` and append the output here, please?

**Comment by kotnik** [comment]
Here is the output you requested:

```
[root@carpo ~]# ls -al /var/log/journal/*/
/var/log/journal/61f38b18f5dac4c7b39724fb0000011a/:
total 76176
drwxr-sr-x+ 2 root systemd-journal    40960 Sep  4 09:31 .
drwxr-sr-x+ 4 root systemd-journal     4096 Jul  3 10:11 ..
-rw-r-----+ 1 root systemd-journal  8388608 Sep  4 10:55 system.journal
-rw-r-----+ 1 root systemd-journal 10485760 Sep  3 15:06 system@054cb3e04ca842aea8de42fac45bb382-0000000000242e33-00051ed42e1dbd16.journal
-rw-r-----+ 1 root systemd-journal 10485760 Sep  3 17:10 system@054cb3e04ca842aea8de42fac45bb382-00000000002460a9-00051ed772419dae.journal
-rw-r-----+ 1 root systemd-journal 10485760 Sep  4 09:31 system@054cb3e04ca842aea8de42fac45bb382-00000000002486f9-00051ed92f75c5f1.journal
-rw-r-x---+ 1 root systemd-journal  8388608 Sep  4 09:33 user-1000.journal
-rw-r-x---+ 1 root systemd-journal  8388608 Sep  3 11:12 user-1000@7362b7cd726e4420a62de76e26aaf1c1-0000000000240e7a-00051ec5d939ace1.journal
-rw-r-x---+ 1 root systemd-journal  8388608 Sep  4 09:31 user-1000@7362b7cd726e4420a62de76e26aaf1c1-000000000024874f-00051ed93071d541.journal
-rw-r-x---+ 1 root systemd-journal  8388608 Aug 22 01:38 user-1001.journal
-rwxr-xr-x+ 1 root systemd-journal   196608 Dec 25  2012 user-1002.journal
-rwxr-xr-x+ 1 root systemd-journal  4313088 Oct 31  2012 user-120.journal

/var/log/journal/remote/:
total 16
drwxr-sr-x+ 2 root systemd-journal-remote 4096 Apr 20 10:07 .
drwxr-sr-x+ 4 root systemd-journal        4096 Jul  3 10:11 ..
```

**Comment by poettering** [comment]
Well, the vacuuming only looks at archived files and corrupted files, it will however never look at active files. Now, in your output above the files with "@" in them are the archived ones. However, systemd.journal and the four user-XYZ.journal are still active, and they are not cleaned up hence.

I figure we should provide a way to rotate and then vacuum really _all_ unit files. We currently do not provide that. Will change the bug title accordingly.

**Comment by poettering** [comment]
Fix is waiting in #10525.

**Comment by afreof** [comment]
My conclusion is that journald rotates and vacuums as expected. journald-remote rotates but never vacuums. On a 100MB log partition /var/log/journal stays at ~10MB. /var/log/journal/remote fills the disk with 1MB files.

One related function is here: [server_vacuum ](https://github.com/systemd/systemd/blob/37b8602fb8453aa9bc29d5b6add36a8a56a02705/src/journal/journald-server.c#L690). This function handles files in /(var¦run)/log/journal.
I'm not able to find something similar for journald-remote in the code.

Is is possible to configure journald-remote to log to /var/log/journal to get journald vacuuming these as well? Or is it risky?
